### ISSUE TITLE
Retention policy on backup with many sets throws Exception

### ISSUE THREAD

**Issue by mnaiman** [duplicati/duplicati#3308]
- [x ] I have searched open and closed issues for duplicates.

----------------------------------------

- **Duplicati version**: 2.0.3.9
- **Operating system**: WIN
- **Backend**: FILE

## Description
After applying retention-policy Duplicati wants to delete 1149 filesets from backup (correct).
 
This code is not ready for so many parameters and Exception is thrown.
https://github.com/duplicati/duplicati/blob/e70f15287e77708a475471a5a24596ac2fa79f13/Duplicati/Library/Main/Database/LocalDeleteDatabase.cs#L75

```
2018-07-04 20:17:12 +02 - [Information-Duplicati.Library.Main.Operation.DeleteHandler-DeleteRemoteFileset]: Deleting 1149 remote fileset(s) ...
2018-07-04 20:17:12 +02 - [Error-Duplicati.Library.Main.Operation.BackupHandler-FatalError]: Fatal error
System.Data.SQLite.SQLiteException (0x80004005): SQL logic error or missing database
too many SQL variables
   at System.Data.SQLite.SQLite3.Prepare(SQLiteConnection cnn, String strSql, SQLiteStatement previous, UInt32 timeoutMS, String& strRemain)
   at System.Data.SQLite.SQLiteCommand.BuildNextCommand()
   at System.Data.SQLite.SQLiteDataReader.NextResult()
   at System.Data.SQLite.SQLiteDataReader..ctor(SQLiteCommand cmd, CommandBehavior behave)
   at System.Data.SQLite.SQLiteCommand.ExecuteReader(CommandBehavior behavior)
   at System.Data.SQLite.SQLiteCommand.ExecuteNonQuery(CommandBehavior behavior)
   at Duplicati.Library.Main.Database.ExtensionMethods.ExecuteNonQuery(IDbCommand self, String cmd, Object[] values)
   at Duplicati.Library.Main.Database.LocalDeleteDatabase.<DropFilesetsFromTable>d__6.MoveNext()
   at System.Linq.Buffer`1..ctor(IEnumerable`1 source)
   at System.Linq.Enumerable.ToArray[TSource](IEnumerable`1 source)
   at Duplicati.Library.Main.Operation.DeleteHandler.DoRun(LocalDeleteDatabase db, IDbTransaction& transaction, Boolean hasVerifiedBacked, Boolean forceCompact, BackendManager sharedManager)
   at Duplicati.Library.Main.Operation.BackupHandler.CompactIfRequired(BackendManager backend, Int64 lastVolumeSize)
   at Duplicati.Library.Main.Operation.BackupHandler.<RunAsync>d__19.MoveNext()
```

**Comment by mnaiman** [comment]
https://www.sqlite.org/limits.html
9. Maximum Number Of Host Parameters In A Single SQL Statement
To prevent excessive memory allocations, the maximum value of a host parameter number is SQLITE_MAX_VARIABLE_NUMBER, which defaults to 999

**Comment by warwickmm** [comment]
Fixed in pull request #3308.
